be/src/olap/delta_writer.cpp

```diff
@@ -291,17 +305,19 @@ void DeltaWriter::_request_slave_tablet_pull_rowset(const PNodeInfo& node_info)
                     std::string inverted_index_file =
                             InvertedIndexDescriptor::get_index_file_path_v1(
                                     index_path_prefix, index_meta.first, index_meta.second);
-                    int64_t size = std::filesystem::file_size(inverted_index_file);
-                    PTabletWriteSlaveRequest::IndexSize index_size;
-                    index_size.set_indexid(index_meta.first);
-                    index_size.set_size(size);
-                    index_size.set_suffix_path(index_meta.second);
-                    // Fetch the map value for the current segment_id.
-                    // If it doesn't exist, this will insert a new default-constructed IndexSizeMapValue
-                    auto& index_size_map_value =
-                            (*(request->mutable_inverted_indices_size()))[segment_id];
-                    // Add the new index size to the map value.
-                    *index_size_map_value.mutable_index_sizes()->Add() = std::move(index_size);
+                    int64_t size = safe_get_file_size(inverted_index_file);
```

Comment:
Should return directly instead of not executing the logic if there is an exception?